test/Microsoft.AspNet.Mvc.Localization.Test/MvcLocalizationServiceCollectionExtensionsTest.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.AspNet.Mvc.Razor;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Framework.WebEncoders;
+using Microsoft.Framework.WebEncoders.Testing;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Localization.Test
+{
+    public class MvcLocalizationServiceCollectionExtensionsTest
+    {
+        [Fact]
+        public void AddMvcLocalization_AddsNeededServices()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+
+            // Act
+            MvcLocalizationServiceCollectionExtensions.AddMvcLocalization(collection);
+
+            // Assert
+            var services = collection.ToList();
+            Assert.Equal(7, services.Count);
+
+            Assert.Equal(typeof(IConfigureOptions<RazorViewEngineOptions>), services[0].ServiceType);
+            Assert.Equal(ServiceLifetime.Singleton, services[0].Lifetime);
+
+            Assert.Equal(typeof(IHtmlLocalizerFactory), services[1].ServiceType);
+            Assert.Equal(typeof(HtmlLocalizerFactory), services[1].ImplementationType);
+            Assert.Equal(ServiceLifetime.Singleton, services[1].Lifetime);
+
+            Assert.Equal(typeof(IHtmlLocalizer<>), services[2].ServiceType);
+            Assert.Equal(typeof(HtmlLocalizer<>), services[2].ImplementationType);
+            Assert.Equal(ServiceLifetime.Transient, services[2].Lifetime);
+
+            Assert.Equal(typeof(IViewLocalizer), services[3].ServiceType);
+            Assert.Equal(typeof(ViewLocalizer), services[3].ImplementationType);
+            Assert.Equal(ServiceLifetime.Transient, services[3].Lifetime);
+
+            Assert.Equal(typeof(IHtmlEncoder), services[4].ServiceType);
+            Assert.Equal(ServiceLifetime.Singleton, services[4].Lifetime);
+
+            Assert.Equal(typeof(IStringLocalizerFactory), services[5].ServiceType);
+            Assert.Equal(typeof(ResourceManagerStringLocalizerFactory), services[5].ImplementationType);
+            Assert.Equal(ServiceLifetime.Singleton, services[5].Lifetime);
+
+            Assert.Equal(typeof(IStringLocalizer<>), services[6].ServiceType);
+            Assert.Equal(typeof(StringLocalizer<>), services[6].ImplementationType);
+            Assert.Equal(ServiceLifetime.Transient, services[6].Lifetime);
+        }
+
+        [Fact]
+        public void AddCustomLocalizers_BeforeMvcLocalization()
```

Comment:
This and next test name do not describe the expectations. Should end with "_AddsNeededServices" or (better) something more specific.